src/neo/SmartContract/ApplicationEngine.OpCodePrices.cs

```diff
@@ -7,190 +7,190 @@ partial class ApplicationEngine
-            [OpCode.REVERSEITEMS] = 4096,
-            [OpCode.REMOVE] = 64,
-            [OpCode.CLEARITEMS] = 512,
-            [OpCode.ISNULL] = 2,
-            [OpCode.ISTYPE] = 4,
-            [OpCode.CONVERT] = 64,
+            [OpCode.DEPTH] = 1 << 1,
+            [OpCode.DROP] = 1 << 1,
+            [OpCode.NIP] = 1 << 1,
+            [OpCode.XDROP] = 1 << 4,
+            [OpCode.CLEAR] = 1 << 4,
+            [OpCode.DUP] = 1 << 1,
+            [OpCode.OVER] = 1 << 1,
+            [OpCode.PICK] = 1 << 1,
+            [OpCode.TUCK] = 1 << 1,
+            [OpCode.SWAP] = 1 << 1,
+            [OpCode.ROT] = 1 << 1,
+            [OpCode.ROLL] = 1 << 4,
+            [OpCode.REVERSE3] = 1 << 1,
+            [OpCode.REVERSE4] = 1 << 1,
+            [OpCode.REVERSEN] = 1 << 4,
+            [OpCode.INITSSLOT] = 1 << 4,
+            [OpCode.INITSLOT] = 1 << 6,
+            [OpCode.LDSFLD0] = 1 << 1,
+            [OpCode.LDSFLD1] = 1 << 1,
+            [OpCode.LDSFLD2] = 1 << 1,
+            [OpCode.LDSFLD3] = 1 << 1,
+            [OpCode.LDSFLD4] = 1 << 1,
+            [OpCode.LDSFLD5] = 1 << 1,
+            [OpCode.LDSFLD6] = 1 << 1,
+            [OpCode.LDSFLD] = 1 << 1,
+            [OpCode.STSFLD0] = 1 << 1,
+            [OpCode.STSFLD1] = 1 << 1,
+            [OpCode.STSFLD2] = 1 << 1,
+            [OpCode.STSFLD3] = 1 << 1,
+            [OpCode.STSFLD4] = 1 << 1,
+            [OpCode.STSFLD5] = 1 << 1,
+            [OpCode.STSFLD6] = 1 << 1,
+            [OpCode.STSFLD] = 1 << 1,
+            [OpCode.LDLOC0] = 1 << 1,
+            [OpCode.LDLOC1] = 1 << 1,
+            [OpCode.LDLOC2] = 1 << 1,
+            [OpCode.LDLOC3] = 1 << 1,
+            [OpCode.LDLOC4] = 1 << 1,
+            [OpCode.LDLOC5] = 1 << 1,
+            [OpCode.LDLOC6] = 1 << 1,
+            [OpCode.LDLOC] = 1 << 1,
+            [OpCode.STLOC0] = 1 << 1,
+            [OpCode.STLOC1] = 1 << 1,
+            [OpCode.STLOC2] = 1 << 1,
+            [OpCode.STLOC3] = 1 << 1,
+            [OpCode.STLOC4] = 1 << 1,
+            [OpCode.STLOC5] = 1 << 1,
+            [OpCode.STLOC6] = 1 << 1,
+            [OpCode.STLOC] = 1 << 1,
+            [OpCode.LDARG0] = 1 << 1,
+            [OpCode.LDARG1] = 1 << 1,
+            [OpCode.LDARG2] = 1 << 1,
+            [OpCode.LDARG3] = 1 << 1,
+            [OpCode.LDARG4] = 1 << 1,
+            [OpCode.LDARG5] = 1 << 1,
+            [OpCode.LDARG6] = 1 << 1,
+            [OpCode.LDARG] = 1 << 1,
+            [OpCode.STARG0] = 1 << 1,
+            [OpCode.STARG1] = 1 << 1,
+            [OpCode.STARG2] = 1 << 1,
+            [OpCode.STARG3] = 1 << 1,
+            [OpCode.STARG4] = 1 << 1,
+            [OpCode.STARG5] = 1 << 1,
+            [OpCode.STARG6] = 1 << 1,
+            [OpCode.STARG] = 1 << 1,
+            [OpCode.NEWBUFFER] = 1 << 11,
+            [OpCode.MEMCPY] = 1 << 11,
+            [OpCode.CAT] = 1 << 11,
+            [OpCode.SUBSTR] = 1 << 11,
+            [OpCode.LEFT] = 1 << 11,
+            [OpCode.RIGHT] = 1 << 11,
```

Comment:
In general CAT/SUBSTR/LEFT/RIGHT tend to cost more than MEMPCY.

Reply:
agree